agfusion/parsers.py

```diff
@@ -24,7 +26,13 @@ def __next__(self):
 
     def _check_data(self):
         if len(self.fusions) == 0:
-            self.logger.error("Read 0 fusions from the file! Exiting...")
+            #self.logger.error("Read 0 fusions from the file! Exiting...")
+            if not os.path.exists("agfusion_results"):
+                os.mkdir("agfusion_results")
+            with zipfile.ZipFile("/cromwell_root/agfusion_results.zip", mode="w") as archive:
```

Comment:
If AGFusion is run outside of cromwell this seems likely to break.  But I don't think we should be making this file here.  AGFusion doesn't make a ZIP file of its results normally.  If one is needed, that should be created by the workflow step, just as it currently is in the case where results exist.

Reply:
I agree that hardcoding an expectation of`/cromwell_root/ ...` is probably not what we want 

Two comments.
1.) As far as I know this fork of AGfusion only exists for us to run workflows
2.) I'm not sure we would follow your last suggestion though. I think by the time we know we need to create an empty result, AGfusion has already died, and this kills the workflow...

Reply:
Is there a way that we can catch the failure of AGfusion here?:
https://github.com/griffithlab/analysis-wdls/blob/018f5c0721252a57c22435a68123f611612fb7f1/definitions/tools/agfusion.wdl#L18-L25 

And if that command did not succeed, instead of going into the output dir and zipping up the results, we just create an empty result right there in that block in the WDL?

Reply:
That has its own problems in my mind.  We don't want to do this for *any* failure of AGfusion.  We wanted it to be somehow specific to this case where AGfusion tells us. I found nothing to annotate.

Reply:
What if we let AGfusion create an empty results dir at this point but do NOT create the zip file?  Let that part happen in the WDL as is it would normally?

Reply:
And instead of creating an empty folder with hard coded name, we use the output dir name that passed into AGfusion as a param.

Reply:
> 2.) I'm not sure we would follow your last suggestion though. I think by the time we know we need to create an empty result, AGfusion has already died, and this kills the workflow...

I agree that we can have the tool make an empty result as needed, but the ZIP file specifically is never created by AGFusion. It is created here normally:
https://github.com/griffithlab/analysis-wdls/blob/018f5c0721252a57c22435a68123f611612fb7f1/definitions/tools/agfusion.wdl#L24
It doesn't make sense to instead make it here in the case that there are no fusions.  That's not a responsibility of this tool.